Bluetooth: Log refactor (cont.)

theob-pro opened the pull request:
Last part of the log refactor work.

The `common/log` has been removed and file that were using it for log has been updated.
More details in the commit message.

theob-pro committed:
Bluetooth: Use Zephyr standard log system instead of `bluetooth/common/log`

The `bluetooth/common/log.h` and `bluetooth/common/log.c` files have been
removed. Files that were using them have been updated to use
`zephyr/logging/log.h` instead.

Those replacement have been done consequently:
- `/BT_DBG/LOG_DBG/`
- `/BT_ERR/LOG_ERR/`
- `/BT_WARN/LOG_WRN/`
- `/BT_INFO/LOG_INF/`
- `/BT_HEXDUMP_DBG/LOG_HEXDUMP_DBG/`
- `/BT_DBG_OBJ_ID/LOG_DBG_OBJ_ID/`

Also, some files were relying on the `common/log.h` include to include
`zephyr/bluetooth/hci.h`, in those cases the include of `hci.h` has
been added.

For files that were including `common/log.h` but not using any logs,
the include has been removed and not replaced.

Signed-off-by: Théo Battrel <theo.battrel@nordicsemi.no>

LingaoM commented on drivers/bluetooth/hci/h5.c:
```diff
@@ -566,7 +576,7 @@ static void bt_uart_isr(const struct device *unused, void *user_data)
 			    H5_HDR_SEQ(hdr) != h5.tx_ack) {
-				BT_ERR("Seq expected %u got %u. Drop packet",
+				LOG_ERR("Seq expected %u got %u. Drop packet",
 				       h5.tx_ack, H5_HDR_SEQ(hdr));
```
Maybe add space here better for alignment

Thalley replied:
This will be a general comment if it is not done everywhere:

For all multiline `BT_<XXX>" statements, the next line should have been indented by 7 spaces. With `LOG_<XXX>", this indent is now 8 spaces, which shall be converted to a tab.

theob-pro replied:
Good point. I thought git-clang-format would fix this. But I fell into a trap: The second line there was not modified, so it was not touched by git-clang-format. Fixed now.

jhedberg commented on drivers/bluetooth/hci/h4.c:
```diff
@@ -22,9 +22,19 @@
-#define BT_DBG_ENABLED IS_ENABLED(CONFIG_BT_DEBUG_HCI_DRIVER)
-#define LOG_MODULE_NAME bt_driver
-#include "common/log.h"
+#include <zephyr/logging/log.h>
+
+#ifdef CONFIG_BT_DEBUG_LOG
+#ifdef CONFIG_BT_DEBUG_HCI_DRIVER
+#define LOG_LEVEL LOG_LEVEL_DBG
+#else
+#define LOG_LEVEL LOG_LEVEL_INF
+#endif
+#else
+#define LOG_LEVEL LOG_LEVEL_NONE
+#endif
+
+LOG_MODULE_REGISTER(bt_driver, LOG_LEVEL);
```
Since you have to repeat these 13 lines in every single c-file, maybe it would still make sense to keep a `common/log.h` header with a helper for this, e.g.:

```c
#include "common/log.h"
BT_LOG_REGISTER(bt_driver, IS_ENABLED(CONFIG_BT_DEBUG_HCI_DRIVER));
```

The `BT_LOG_REGISTER()` API in `common/log.h` could be defined something like (completely untested):
```c
#ifdef CONFIG_BT_DEBUG_LOG
#define BT_LOG_LEVEL_1 LOG_LEVEL_DBG
#define BT_LOG_LEVEL_0 LOG_LEVEL_INF
#else
#define BT_LOG_LEVEL_0 LOG_LEVEL_NONE
#endif

#define BT_LOG_REGISTER(name, enable) \
	LOG_MODULE_REGISTER(name, _CONCAT(BT_LOG_LEVEL_, enable))
```

Thalley replied:
Generally in BT we've had something similar to 
```c
#define BT_DBG_ENABLED IS_ENABLED(CONFIG_BT_DEBUG_SMP)
#define LOG_MODULE_NAME bt_smp
#include "common/log.h"
```

Which I guess could be transformed into something a bit shorter like
```c
#include <zephyr/logging/log.h>

LOG_MODULE_REGISTER(bt_smp, IS_ENABLED(CONFIG_BT_DEBUG_SMP) ? LOG_LEVEL_DBG : CONFIG_BT_LOG_LEVEL);
```

The alternative is to actually define a log level for each file/module instead.

jhedberg replied:
@Thalley are you sure `LOG_MODULE_REGISTER()` accepts a conditional statement as its second argument? I assumed it needs to be a directly valid preprocessor symbol, which is what I based my proposal on.

alwa-nordic replied:
Zephyr has a preprocessor ternary: `COND_CODE_1`.

Thalley replied:
It would appear not. It doesn't fail at building, but it does fail at runtime. 

In that case, your suggestion is possibly better, but perhaps mine could be modified to

```c
#include <zephyr/logging/log.h>
#if defined(CONFIG_BT_DEBUG_SMP)
#define SMP_LOG_LEVEL LOG_LEVEL_DBG
#else 
#define SMP_LOG_LEVEL CONFIG_BT_LOG_LEVEL
#endif

LOG_MODULE_REGISTER(bt_smp, SMP_LOG_LEVEL);
```

Which does work, but isn't much better than what @theob-pro is suggesting

alwa-nordic replied:
You can write it like the following:

```c
LOG_MODULE_REGISTER(bt_smp, COND_CODE_1(CONFIG_BT_DEBUG_SMP, (LOG_LEVEL_DBG), (CONFIG_BT_LOG_LEVEL)));
```

Note that @theob-pro's code is a true refactor. Our suggestion here changes the meaning of the kconfigs slightly. (But I don't think that's necessarily an issue.)

Thalley replied:
> Note that @theob-pro's code is a true refactor. Our suggestion here changes the meaning of the kconfigs slightly. (But I don't think that's necessarily an issue.)

Is it though? 

The current code uses 

```c
#define BT_DBG_ENABLED IS_ENABLED(CONFIG_BT_DEBUG_HCI_DRIVER)
#define LOG_MODULE_NAME bt_driver
#include "common/log.h"
```

and in log.h we have

```c
#if BT_DBG_ENABLED
#define LOG_LEVEL LOG_LEVEL_DBG
#else
#define LOG_LEVEL CONFIG_BT_LOG_LEVEL
#endif
```

Wouldn't that indicate that `CONFIG_BT_DEBUG_HCI_DRIVER=y` => `LOG_LEVEL_DBG` and `CONFIG_BT_DEBUG_HCI_DRIVER=n` => `CONFIG_BT_LOG_LEVEL` which is also what I am suggesting. 

The code suggested by @theob-pro 
```c

#ifdef CONFIG_BT_DEBUG_LOG
#ifdef CONFIG_BT_DEBUG_HCI_DRIVER
#define LOG_LEVEL LOG_LEVEL_DBG
#else
#define LOG_LEVEL LOG_LEVEL_INF
#endif
#else
#define LOG_LEVEL LOG_LEVEL_NONE
#endif
```

Sets `LOG_LEVEL` to `LOG_LEVEL_INF` if `CONFIG_BT_DEBUG_HCI_DRIVER=n` which is different that the current code (it should be `CONFIG_BT_LOG_LEVEL`). 

Or am I missing something?

alwa-nordic replied:
Ah. I didn't see the definition of `config BT_LOG_LEVEL` because it's instantiated from a template.

I see now that you are correct. @Thalley's proposal is the correct refactoring. Agree, @theob-pro?

Thalley commented on samples/bluetooth/iso_broadcast_benchmark/src/broadcaster.c:
```diff
@@ -411,8 +410,7 @@ static int create_big(struct bt_le_ext_adv **adv, struct bt_iso_big **big)
 	if (err != 0) {
-		LOG_ERR("Failed to set periodic advertising parameters (err %d)",
-			err);
+		LOG_ERR("Failed to set periodic advertising parameters (err %d)", err);
```
The changes in this file seems unnecessary.

Thalley replied:
Hmm, this seems to be the case for several files. Are you using some kind of auto-format that is modifying lines of code, just to make them a single line? 

If so, please turn that off, and revert those changes.

jori-nordic replied:
These changes stem from this comment: https://github.com/zephyrproject-rtos/zephyr/pull/51897#discussion_r1011809309
I don't think it's reasonable to ask him to format all of those changes manually. 

He's running clang-format on his changes with the zephyr rules, if you don't like the formatting, please update the rules or provide an automated tool to do it.

This might seem like a bad-faith argument from me, since clang-format has some limitations, but I really think those nitpicky formatting comments (by you, me or anyone else) shouldn't be necessary and should be encoded in the CI checks, and if they can't, we should refrain from commenting/requesting changes.

I haven't used it a lot, but maybe coccinelle can do this sort of thing? pinging @carlescufi if he knows, since that's a recurring issue in PRs.

alwa-nordic replied:
I agree! We really need to just get everything clang-formatted. Formatting issues are consistently a big impediment to merging PRs in Zephyr. That's a waste of everyone's time in my opinion.

Thalley replied:
> He's running clang-format on his changes with the zephyr rules, if you don't like the formatting, please update the rules or provide an automated tool to do it.

If that was the case, then that would be fine, but this case (and many more) is not changed by his PR, except to remove this line brake. 

This PR is disruptive enough already (I'll have to rebase dozens of commits after this already) without modifying lines just for the sake of modifying it. 

Besides this specific line also makes the line longer than 80 characters, which is the recommended maximum. 

To me it seems that he has incorrectly configured his clang-format, and neither of the following should occur in this PR:
1) We should not remove newlines the same of removing newlines if the code already follows the code standards (which it did). 
2) We should not automatically make any line exceed the 80 character recommended maximum, unless it significantly improved readability. 

IMO this PR breaks our current coding standards.

theob-pro commented on subsys/bluetooth/mesh/net.c:
```diff
@@ -493,7 +501,7 @@ static int loopback(const struct bt_mesh_net_tx *tx, const uint8_t *data,
 	err = k_mem_slab_alloc(&loopback_buf_pool, (void **)&buf, K_NO_WAIT);
 	if (err) {
-		BT_WARN("Unable to allocate loopback");
+		LOG_WRN("Unable to allocate loopback");
```
@stephanosio The CI is failing because of a compliance failure (`WARNING:EMBEDDED_FUNCTION_NAME`) but this is a false positive. The same is happening with this file: `subsys/bluetooth/host/settings.c:191`. 
Is there a way to fix that?

stephanosio replied:
well, a way to "fix" that would be to use a less generic name for the function (e.g. `send_loopback`, `do_loopback`, ...).

The `WARNING:EMBEDDED_FUNCTION_NAME` is often useful and I do not think we should disable it. If there are false positives, then there are false positives. We just need to override the checks when merging.

Thalley replied:
Indeed. A common (no-brainer) prefix for such functions is to simply add the module name as the prefix, i.e. `mesh_loopback` in this case.

LingaoM replied:
Well, I remember that the same problem happened in the past, and the discussion at that time was that global functions or any public symbols need to be prefixed, and I think the same applies to this problem, but it is indeed a huge project, but I think it should be  Form a rule to monitor.

theob-pro replied:
I changed the functions' names. I welcome better ideas for names.

theob-pro commented:
Reworked the PR to be consistent with the other subsystems by using Kconfig logging template to define `CONFIG_*_LOG_LEVEL`.

nordic-krch commented on include/zephyr/bluetooth/audio/tbs.h:
```diff
@@ -326,10 +326,10 @@ int bt_tbs_set_uri_scheme_list(uint8_t bearer_index, const char **uri_list,
-#if defined(CONFIG_BT_DEBUG_TBS)
+#if defined(CONFIG_BT_TBS_LOG_LEVEL_DBG)
 /** @brief Prints all calls of all services to the debug log */
 void bt_tbs_dbg_print_calls(void);
```
I would check if you need those ifdefs at all. When debug logs are off function is empty and never called. I would expect that linker handles it.

theob-pro replied:
Fixing this is not part of this PR.

nordic-krch commented on subsys/bluetooth/controller/ll_sw/openisa/hal/RV32M1/ecb.c:
```diff
@@ -8,16 +8,20 @@
 #include "hal/debug.h"
 #include "fsl_cau3_ble.h"
 
+LOG_MODULE_REGISTER(bt_ctlr_rv32m1_ecb, LOG_LEVEL_DBG);
```
shouldn't here be a level from kconfig?

theob-pro replied:
Not really if we look at how it was done previously. There was no `BT_DBG_ENABLED` so the default behavior was to use `LOG_LEVEL_DBG`. As it's a refactor PR I don't think that I should change that behavior.

You can check by yourself with that:
```c
#define XSTR(x) STR(x)
#define STR(x) #x
#pragma message "Log level: " XSTR(LOG_LEVEL)
```

nordic-krch commented on subsys/bluetooth/controller/ll_sw/openisa/hal/RV32M1/radio/radio.c:
```diff
@@ -26,13 +26,15 @@
 
+#define LOG_LEVEL CONFIG_BT_HCI_DRIVER_LOG_LEVEL
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(bt_openisa_radio);
```
no need to define `LOG_LEVEL` just put it as second argument to the macro.

theob-pro replied:
This won't work when `CONFIG_BT_HCI_DRIVER_LOG_LEVEL` is not defined, which unfortunately can happen right now.

If `CONFIG_BT_HCI_DRIVER_LOG_LEVEL` is placed as the second argument of `LOG_MODULE_REGISTER` but `CONFIG_BT_HCI_DRIVER_LOG_LEVEL` is undefined, it will result in `LOG_MODULE_REGISTER` having an empty second argument which is not allowed.

It's already done like that in others subsystems, here are some examples:
https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/usb/device/os_desc.c#L7
https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/net/buf.c#L10
https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/net/lib/lwm2m/lwm2m_message_handling.c#L16
https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/emul/i2c/emul_sbs_gauge.c#L11

nordic-krch replied:
I see, so there is a file that is compiled but it uses a level from module which might be disabled (and because of that level is not available)? It looks as if some dependency cleanup would be needed but as i understand this PR does not intend to change anything.

nordic-krch commented on subsys/bluetooth/controller/ll_sw/ull_conn.c:
```diff
@@ -70,11 +70,12 @@
 
+#define LOG_LEVEL CONFIG_BT_HCI_DRIVER_LOG_LEVEL
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(bt_ctlr_ull_conn);
```
log level as second argument.

nordic-krch reviewed (commented):
There are multiple places where `LOG_LEVEL` is defined but it could be just used as second argument to `LOG_MODULE_REGISTER` (like it's done in multiple other places).

Thalley commented:
Just FYI I won't be reviewing this PR again until after the upcoming UPF (end December 8th). 

Feel free to merge it without my approval though.

theob-pro commented:
> Just FYI I won't be reviewing this PR again until after the upcoming UPF (end December 8th).
> 
> Feel free to merge it without my approval though.

@Thalley Could you please dismiss your change request then? The only updated lines are now because of the macro change.

jhedberg commented on drivers/bluetooth/hci/h4.c:
```diff
@@ -22,9 +22,10 @@
-#define BT_DBG_ENABLED IS_ENABLED(CONFIG_BT_DEBUG_HCI_DRIVER)
-#define LOG_MODULE_NAME bt_driver
-#include "common/log.h"
+#define LOG_LEVEL CONFIG_BT_HCI_DRIVER_LOG_LEVEL
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(bt_driver);
```
Can't these be shortened into:
```c
#include <zephyr/logging/log.h>
LOG_MODULE_REGISTER(bt_driver, CONFIG_BT_HCI_DRIVER_LOG_LEVEL);
```

Same for all other similar places.

jhedberg replied:
Right, I see.

theob-pro commented:
Rebased to fix CI. (`subsys/bluetooth/controller/ll_sw/nordic/lll/lll_central_iso.c` has been added and were using `common/log.h`)

carlescufi reviewed (changes requested):
No objections in principle, though this is effectively an API change, because the way logging is enabled for different sub-subsystems within Bluetooth has been modified. I would ask at the very least you add this to the 3.3 release notes.

Also I have a question. Before, all `BT_WARN` and `BT_ERR` messages would display by default. Is this still the case for all `LOG_WRN` and `LOG_ERR` in the different modules within Bluetooth?

theob-pro commented:
Rebased to fix conflicting files: `subsys/bluetooth/audio/mcc.c`.
Also added a test (see the last commit) and fixed this file: `drivers/bluetooth/hci/userchan.c`. This last change has been squashed into the first commit.

theob-pro commented:
Rebased to fix conflicting files: `subsys/bluetooth/audio/mcc.c`.
Also added this work to the release note as requested. It's in the last commit.

jhedberg commented on drivers/bluetooth/hci/spi.c:
```diff
@@ -17,9 +17,9 @@
-#include "common/log.h"
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(bt_driver, CONFIG_BT_HCI_DRIVER_LOG_LEVEL);
```
Why is it here ok to pass the Kconfig option directly to `LOG_MODULE_REGISTER()` but not in other places?

jori-nordic replied:
conditional inclusion of the log template in kconfig:
if the template is if'd out, (e.g. because of `BT_SMP` not being set), the `SMP` log options won't exist and `LOG_MODULE_REGISTER` will not work since the define will not exist.

in this case, the template is not excluded depending on some kconfig, so it works

theob-pro replied:
This is a mistake for this case. I took a look again at all the file and fixed the same mistake on 7 others files.

Let me add that some files like `subsys/bluetooth/host/l2cap.c` pass the Kconfig option directly to `LOG_MODULE_REGISTER` because the Kconfig is not depending on Kconfig that could be disabled and so their Kconfig log symbol is always defined.

theob-pro commented:
Rebased to fix conflicting file: `subsys/bluetooth/mesh/transport.c`.
Also updated the release note section concerning this work.

theob-pro commented:
Rebased to fix conflicting files: 
`subsys/bluetooth/audio/broadcast_source.c`;
`subsys/bluetooth/audio/stream.c`;
`tests/bluetooth/bsim_bt/bsim_test_mesh/prj_pst.conf`.

Also added back Kconfig symbols that were used for Bluetooth logging subsystem to avoid breaking the stable API. More details in the last commit.

theob-pro commented:
Rebased to fix conflicting files:
` subsys/bluetooth/audio/Kconfig.csis `;
` subsys/bluetooth/audio/cap_acceptor.c `;
` subsys/bluetooth/audio/csis.c `;
` subsys/bluetooth/audio/csis_client.c `;
` subsys/bluetooth/audio/csis_crypto.c `;
` tests/bluetooth/bsim_bt/bsim_test_audio/prj.conf `;
` tests/bluetooth/shell/audio.conf `.

theob-pro commented:
Rebased to fix conflicting file: ` subsys/bluetooth/mesh/proxy_srv.c `.

tejlmand commented on subsys/bluetooth/audio/Kconfig.aics:
```diff
@@ -34,10 +34,17 @@ config BT_AICS_MAX_INPUT_DESCRIPTION_SIZE
+legacy-debug-sym = BT_DEBUG_AICS
 module-str = "Audio Input Control Service"
-source "${ZEPHYR_BASE}/subsys/logging/Kconfig.template.log_config_inherit"
+source "${ZEPHYR_BASE}/subsys/bluetooth/common/Kconfig.template.log_config_bt"
```
minor nit, `${ZEPHYR_BASE}` can be omitted when using `source`.

```suggestion
source "subsys/bluetooth/common/Kconfig.template.log_config_bt"
```

carlescufi replied:
This should’ve been caught by compliance I think. I will check why it wasn’t.

tejlmand commented on subsys/bluetooth/audio/Kconfig.aics:
```diff
@@ -62,9 +69,16 @@ if BT_AICS_CLIENT
+legacy-debug-sym = BT_DEBUG_AICS_CLIENT
 module-str = "Audio Input Control Service client"
-source "${ZEPHYR_BASE}/subsys/logging/Kconfig.template.log_config_inherit"
+source "${ZEPHYR_BASE}/subsys/bluetooth/common/Kconfig.template.log_config_bt"
```
```suggestion
source "subsys/bluetooth/common/Kconfig.template.log_config_bt"
```

tejlmand commented on subsys/bluetooth/audio/Kconfig.baps:
```diff
@@ -260,10 +260,18 @@ config BT_BAP_BROADCAST_ASSISTANT_RECV_STATE_COUNT
+legacy-debug-sym = BT_AUDIO_DEBUG_STREAM
 module-str = "Bluetooth Audio Stream"
-source "${ZEPHYR_BASE}/subsys/logging/Kconfig.template.log_config_inherit"
+source "${ZEPHYR_BASE}/subsys/bluetooth/common/Kconfig.template.log_config_bt"
```
```suggestion
source "subsys/bluetooth/common/Kconfig.template.log_config_bt"
```

tejlmand commented on subsys/bluetooth/audio/Kconfig.baps:
```diff
@@ -273,40 +281,96 @@ config BT_AUDIO_DEBUG_STREAM_DATA
+legacy-debug-sym = BT_DEBUG_ASCS
 module-str = "Audio Stream Control Service"
-source "${ZEPHYR_BASE}/subsys/logging/Kconfig.template.log_config_inherit"
+source "${ZEPHYR_BASE}/subsys/bluetooth/common/Kconfig.template.log_config_bt"
```
```suggestion
source "subsys/bluetooth/common/Kconfig.template.log_config_bt"
```

tejlmand commented on subsys/bluetooth/audio/Kconfig.baps:
```diff
@@ -273,40 +281,96 @@ config BT_AUDIO_DEBUG_STREAM_DATA
+legacy-debug-sym = BT_AUDIO_DEBUG_UNICAST_SERVER
 module-str = "Bluetooth Audio Unicast Server"
-source "${ZEPHYR_BASE}/subsys/logging/Kconfig.template.log_config_inherit"
+source "${ZEPHYR_BASE}/subsys/bluetooth/common/Kconfig.template.log_config_bt"
```
```suggestion
source "subsys/bluetooth/common/Kconfig.template.log_config_bt"
```

tejlmand commented on subsys/bluetooth/audio/Kconfig.baps:
```diff
@@ -273,40 +281,96 @@ config BT_AUDIO_DEBUG_STREAM_DATA
+legacy-debug-sym = BT_AUDIO_DEBUG_UNICAST_CLIENT
 module-str = "Basic Audio Profile"
-source "${ZEPHYR_BASE}/subsys/logging/Kconfig.template.log_config_inherit"
+source "${ZEPHYR_BASE}/subsys/bluetooth/common/Kconfig.template.log_config_bt"
```
```suggestion
source "subsys/bluetooth/common/Kconfig.template.log_config_bt"
```

tejlmand commented on subsys/bluetooth/audio/Kconfig.baps:
```diff
@@ -273,40 +281,96 @@ config BT_AUDIO_DEBUG_STREAM_DATA
+legacy-debug-sym = BT_AUDIO_DEBUG_BROADCAST_SOURCE
 module-str = "Bluetooth Audio Broadcast Source"
-source "${ZEPHYR_BASE}/subsys/logging/Kconfig.template.log_config_inherit"
+source "${ZEPHYR_BASE}/subsys/bluetooth/common/Kconfig.template.log_config_bt"
```
```suggestion
source "subsys/bluetooth/common/Kconfig.template.log_config_bt"
```

tejlmand commented on subsys/bluetooth/audio/Kconfig.baps:
```diff
@@ -273,40 +281,96 @@ config BT_AUDIO_DEBUG_STREAM_DATA
+legacy-debug-sym = BT_AUDIO_DEBUG_BROADCAST_SINK
 module-str = "Bluetooth Audio Broadcast Sink"
-source "${ZEPHYR_BASE}/subsys/logging/Kconfig.template.log_config_inherit"
+source "${ZEPHYR_BASE}/subsys/bluetooth/common/Kconfig.template.log_config_bt"
```
```suggestion
source "subsys/bluetooth/common/Kconfig.template.log_config_bt"
```

tejlmand commented on subsys/bluetooth/audio/Kconfig.baps:
```diff
@@ -273,40 +281,96 @@ config BT_AUDIO_DEBUG_STREAM_DATA
+legacy-debug-sym = BT_DEBUG_BAP_BROADCAST_ASSISTANT
 module-str = "Broadcast Audio Scan Service client debug"
-source "${ZEPHYR_BASE}/subsys/logging/Kconfig.template.log_config_inherit"
+source "${ZEPHYR_BASE}/subsys/bluetooth/common/Kconfig.template.log_config_bt"
```
```suggestion
source "subsys/bluetooth/common/Kconfig.template.log_config_bt"
```

tejlmand commented on subsys/bluetooth/audio/Kconfig.cap:
```diff
@@ -26,10 +26,18 @@ config BT_CAP_ACCEPTOR_SET_MEMBER
+legacy-debug-sym = BT_DEBUG_CAP_ACCEPTOR
 module-str = "Common Audio Profile"
-source "${ZEPHYR_BASE}/subsys/logging/Kconfig.template.log_config_inherit"
+source "${ZEPHYR_BASE}/subsys/bluetooth/common/Kconfig.template.log_config_bt"
```
```suggestion
source "subsys/bluetooth/common/Kconfig.template.log_config_bt"
```

tejlmand commented on subsys/bluetooth/audio/Kconfig.csip:
```diff
@@ -49,10 +49,17 @@ config BT_CSIP_SET_MEMBER_MAX_INSTANCE_COUNT
+legacy-debug-sym = BT_DEBUG_CSIP_SET_MEMBER
 module-str = "Coordinated Set Identification Service"
-source "${ZEPHYR_BASE}/subsys/logging/Kconfig.template.log_config_inherit"
+source "${ZEPHYR_BASE}/subsys/bluetooth/common/Kconfig.template.log_config_bt"
```
```suggestion
source "subsys/bluetooth/common/Kconfig.template.log_config_bt"
```

tejlmand commented on subsys/bluetooth/audio/Kconfig.csip:
```diff
@@ -96,14 +103,29 @@ config BT_CSIP_SET_COORDINATOR_ENC_SIRK_SUPPORT
+legacy-debug-sym = BT_DEBUG_CSIP_SET_COORDINATOR
 module-str = "Coordinated Set Identification Profile Set Coordinator"
-source "${ZEPHYR_BASE}/subsys/logging/Kconfig.template.log_config_inherit"
+source "${ZEPHYR_BASE}/subsys/bluetooth/common/Kconfig.template.log_config_bt"
```
```suggestion
source "subsys/bluetooth/common/Kconfig.template.log_config_bt"
```

tejlmand commented on subsys/bluetooth/audio/Kconfig.csip:
```diff
@@ -96,14 +103,29 @@ config BT_CSIP_SET_COORDINATOR_ENC_SIRK_SUPPORT
+legacy-debug-sym = BT_DEBUG_CSIP_SET_MEMBER_CRYPTO
 module-str = "Coordinated Set Identification Profile crypto functions"
-source "${ZEPHYR_BASE}/subsys/logging/Kconfig.template.log_config_inherit"
+source "${ZEPHYR_BASE}/subsys/bluetooth/common/Kconfig.template.log_config_bt"
```
```suggestion
source "subsys/bluetooth/common/Kconfig.template.log_config_bt"
```

tejlmand commented on subsys/bluetooth/audio/Kconfig.has:
```diff
@@ -95,10 +95,16 @@ config BT_HAS_PRESET_NAME_DYNAMIC
+legacy-debug-sym = BT_DEBUG_HAS
 module-str = "Hearing Access Service"
-source "${ZEPHYR_BASE}/subsys/logging/Kconfig.template.log_config_inherit"
+source "${ZEPHYR_BASE}/subsys/bluetooth/common/Kconfig.template.log_config_bt"
```
```suggestion
source "subsys/bluetooth/common/Kconfig.template.log_config_bt"
```

tejlmand commented on subsys/bluetooth/audio/Kconfig.has:
```diff
@@ -112,7 +118,14 @@ config BT_HAS_CLIENT
+legacy-debug-sym = BT_DEBUG_HAS_CLIENT
 module-str = "Hearing Access Service Client"
-source "${ZEPHYR_BASE}/subsys/logging/Kconfig.template.log_config_inherit"
+source "${ZEPHYR_BASE}/subsys/bluetooth/common/Kconfig.template.log_config_bt"
```
```suggestion
source "subsys/bluetooth/common/Kconfig.template.log_config_bt"
```

tejlmand commented on subsys/bluetooth/audio/Kconfig.mcs:
```diff
@@ -19,10 +19,17 @@ config BT_MCS
+legacy-debug-sym = BT_DEBUG_MCS
 module-str = "Media Control Service"
-source "${ZEPHYR_BASE}/subsys/logging/Kconfig.template.log_config_inherit"
+source "${ZEPHYR_BASE}/subsys/bluetooth/common/Kconfig.template.log_config_bt"
```
```suggestion
source "subsys/bluetooth/common/Kconfig.template.log_config_bt"
```

tejlmand commented on subsys/bluetooth/audio/Kconfig.mcs:
```diff
@@ -131,9 +138,16 @@ config BT_MCC_SHELL
+legacy-debug-sym = BT_DEBUG_MCC
 module-str = "Media Control Client"
-source "${ZEPHYR_BASE}/subsys/logging/Kconfig.template.log_config_inherit"
+source "${ZEPHYR_BASE}/subsys/bluetooth/common/Kconfig.template.log_config_bt"
```
```suggestion
source "subsys/bluetooth/common/Kconfig.template.log_config_bt"
```

tejlmand reviewed (changes requested):
please cleanup the use of `source "${ZEPHYR_BASE}/..."`

Rest looks good.

tejlmand commented:
@theob-pro thanks for fixing, but please squash the fix with previous commit, there is no reason to have that particular review comment fix as a independent commit.

tejlmand reviewed (approved):
lgtm
